alchemy/tests/test_alchemy.py

```diff
@@ -302,6 +307,137 @@ def test_annihilated_states(platform_name=None, precision=None):
     if (abs(delta) > MAX_DELTA):
         raise Exception("Maximum allowable difference lambda=1 energy and lambda=0 energy in vacuum and periodic box exceeded (was %.8f kcal/mol; allowed %.8f kcal/mol); test failed." % (delta / unit.kilocalories_per_mole, MAX_DELTA / unit.kilocalories_per_mole))
 
+def check_interacting_energy_components(factory, positions):
+    """Compare full and alchemically-modified system energies by energy component.
+
+    Parameters
+    ----------
+    factory : AbsoluteAlchemicalFactory
+        The factory to test.
+    positions : simtk.openmm.unit.Quantity of dimension [nparticles,3] with units compatible with Angstroms
+        The positions to test.
+
+    """
+    alchemical_state = factory.NoninteractingAlchemicalState()
+    energy_components = factory.getEnergyComponents(alchemical_state, positions, use_all_parameters=False)
+    energy_unit = unit.kilojoule_per_mole
+
+    # Mapping between standard reference system and alchemical system forces to be summed to match
+    force_component_match = {
+        'HarmonicBondForce' : ['unmodified HarmonicBondForce', 'alchemically modified HarmonicBondForce'],
+        'HarmonicAngleForce' : ['unmodified HarmonicAngleForce', 'alchemically modified HarmonicAngleForce'],
+        'PeriodicTorsionForce' : ['unmodified PeriodicTorsionForce', 'alchemically modified PeriodicTorsionForce'],
+    }
+
+    import copy
+    reference_system = copy.deepcopy(factory.reference_system)
+    alchemical_system = copy.deepcopy(factory.alchemically_modified_system)
```

Comment:
Everything from the beginning of the function to here is unused, but I guess you wanted to reformat this to look more similar to `check_noninteracting_energy_components()` and it's still unfinished?